edx_proctoring/signals.py

```diff
@@ -0,0 +1,124 @@
+"edx-proctoring signals"
+from django.db.models.signals import pre_save, post_save, pre_delete
+from django.dispatch import receiver
+from edx_proctoring import models
+from edx_proctoring.backends import get_backend_provider
+
+
+@receiver(pre_save, sender=models.ProctoredExam)
+def check_for_category_switch(sender, instance, **kwargs):  # pylint: disable=unused-argument
```

Comment:
Note: this is the only functional difference. Everything else in the PR is cleanup.